### PR TITLE
[runtime-security] Generate SECL documentation

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR adds generator and scripts to generate:
- a JSON file representing the different fields accessible from a SECL expression
- a markdown file documenting the SECL fields and event types based on this JSON file

Please look at https://docs-staging.datadoghq.com/paulcacheux/secl-fields-generator/security_platform/cloud_workload_security/agent_expressions and https://docs-staging.datadoghq.com/paulcacheux/secl-fields-generator/security_platform/cloud_workload_security/backend for the documentation preview

Related documentation PR: https://github.com/DataDog/documentation/pull/11220

### Motivation

The goal is to keep the SECL documentation up-to-date.

### Describe how to test your changes

The documentation for the SECL language should be accessible and up-to-date.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: docs/cloud-workload-security/agent_expressions.md [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+| `!`                   | File             | Not                                      | 7.27          |
+| `^`                   | File             | Binary not                               | 7.27          |
+| `in [elem1, ...]`     | File             | Element is contained in list             | 7.27          |
+| `not in [elem1, ...]` | File             | Element is not contained in list         | 7.27          |
+| `[~pattern, ...]`     | File             | Regex pattern is (not) contained in list | 7.27          |
+| `=~`                  | File             | String matching                          | 7.27          |
+| `&`                   | File             | Binary and                               | 7.27          |
+| `|`                   | File             | Binary or                                | 7.27          |
+| `&&`                  | File             | Logical and                              | 7.27          |
+| `||`                  | File             | Logical or                               | 7.27          |
```
Comment (n06) [inline docs/cloud-workload-security/agent_expressions.md:65]: All of this should be replaced with Auto-generated content in the "Event Types" section

File: docs/cloud-workload-security/agent_expressions.md [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+- link: "/security_platform/cloud_workload_security/getting_started/"
+  tag: "Documentation"
+  text: "Get started with Datadog Cloud Workload Security"
+---
+
+## Agent expression syntax
+Rules for Cloud Workload Security (CWS) are first evaluated in the Datadog Agent to decide what system activity to collect. This portion of a CWS rule is called the Agent expression. Agent expressions use Datadog's Security Language (SECL). The standard format of a SECL expression is as follows:
+
+```
+<trigger>.<event-attribute> <operator> <value> <event-attribute> ...
```
Comment (n06) [inline docs/cloud-workload-security/agent_expressions.md:15]:
```suggestion
<event-type>.<event-attribute> <operator> <value> <event-attribute> ...
```

File: docs/cloud-workload-security/agent_expressions.md [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+Examples:
+* `T=8` and `width=8` both are in *args_options* for the command `ls -T 8 --width=8`
+* `exec.args_options ~= [ “s=.*\’” ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
+
+### File rights
+
+The *file.rights* attribute can now be used in addition to *file.mode*. *file.mode* can hold values set by the kernel, while the *file.rights* only holds the values set by the user. These rights may be more familiar because they are in the `chmod` commands.
+
+## Event types
+
```
Comment (n06) [inline docs/cloud-workload-security/agent_expressions.md:91]:
```suggestion
Event Types correspond to types of activity seen by the system. The currently supported set of event types are:
```

File: docs/cloud-workload-security/backend.md [hunk 4]
```diff
@@ -0,0 +1,884 @@
+# CWS Event Documentation
+
+The CWS event sent to the backend by the security agent respects the following schema:
+```
+BACKEND_EVENT_SCHEMA = {
```
Comment (n06) [inline docs/cloud-workload-security/backend.md:7]: This should also be collapsable

**Review by n06 — COMMENTED** [review]
Thanks Paul! I have a just a few editorial suggestions

File: docs/cloud-workload-security/agent_expressions.md (hunk 1 above)
Comment (paulcacheux) [reply docs/cloud-workload-security/agent_expressions.md:65]: I'm not sure I understand, the trigger part is auto-generated based on the different event types available

Comment (sarina-dd) [reply docs/cloud-workload-security/agent_expressions.md:69]: Want to confirm: is anything on this page auto-generated? It's showing the same content as what is listed in the file here.

Comment (paulcacheux) [reply docs/cloud-workload-security/agent_expressions.md:69]: Yes `docs/cloud-workload-security/agent_expressions.md` and `docs/cloud-workload-security/backend.md` are completely generated

File: docs/cloud-workload-security/backend.md [hunk 5]
```diff
@@ -0,0 +1,894 @@
+---
+title: CWS Event Documentation
```
Comment (sarina-dd) [inline docs/cloud-workload-security/backend.md:2]:
```suggestion
title: Cloud Workload Security Events
```

File: docs/cloud-workload-security/backend.md [hunk 6]
```diff
@@ -0,0 +1,894 @@
...
+kind: documentation
+description: JSON schema documentation of the CWS backend event
+disable_edit: true
+---
+
+# CWS Event Documentation
+
+The CWS event sent to the backend by the security agent respects the following schema:
```
Comment (sarina-dd) [inline docs/cloud-workload-security/backend.md:10]:
```suggestion
The CWS event sent to the backend by the Agent respects the following schema:
```

**Comment by paulcacheux** [comment]
I made the suggested changes to the template files (this is important).
I agree that a bit more clarity about "what is a CWS backend event" is needed and will try to improve this

File: docs/cloud-workload-security/scripts/templates/backend.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+title: Cloud Workload Security (CWS) Events
+kind: documentation
+description: JSON schema documentation of the CWS backend event
+disable_edit: true
+---
+
+After detecting suspicious activity matching the agent expressions, the Agent will send a log
+to the backend containing a `CWS event`.
+
+This event is used to build signals and its fields can be used to build filters in the web application.
```
Comment (n06) [inline docs/cloud-workload-security/scripts/templates/backend.md:11]:
```suggestion
When activity matches a CWS agent expression, a CWS log will be collected from the system containing all the relevant context about the activity.

This log is sent to Datadog, where it is analyzed. Based on analysis, CWS logs can trigger Security Signals to fire, or they can simply be stored as logs for audit, investigation, or threat hunting purposes.
```

File: docs/cloud-workload-security/scripts/templates/backend.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+The CWS event sent to the backend by the Agent respects the following JSON schema:
```
Comment (n06) [inline docs/cloud-workload-security/scripts/templates/backend.md:13]:
```suggestion
CWS logs have the following JSON schema:
```

File: docs/cloud-workload-security/scripts/templates/agent_expressions.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+description: "Agent expression attributes and operators for Cloud Workload Security Rules"
+disable_edit: true
+further_reading:
+- link: "/security_platform/cloud_workload_security/getting_started/"
+  tag: "Documentation"
+  text: "Get started with Datadog Cloud Workload Security"
+---
+
+## Agent expression syntax
+Rules for Cloud Workload Security (CWS) are first evaluated in the Datadog Agent to decide what system activity to collect. This portion of a CWS rule is called the Agent expression. Agent expressions use Datadog's Security Language (SECL). The standard format of a SECL expression is as follows:
```
Comment (n06) [inline docs/cloud-workload-security/scripts/templates/agent_expressions.md:13]:
```suggestion
Cloud Workload Security (CWS) first evaluates activity within the Datadog Agent against agent expressions to decide what activity to collect. Agent expressions use Datadog's Security Language (SECL). The standard format of a SECL expression is as follows:
```

**Review by n06 — CHANGES_REQUESTED** [review]
Tried to add some editorial changes. @sarina-dd I think this is where you can help us with a better structure/flow

**Review by sarina-dd — APPROVED** [review]
Docs 👍

File: docs/cloud-workload-security/scripts/templates/agent_expressions.md (hunk 9 above)
Comment (sarina-dd) [inline docs/cloud-workload-security/scripts/templates/agent_expressions.md:13]:
```suggestion
Cloud Workload Security (CWS) first evaluates activity within the Datadog Agent against Agent expressions to decide what activity to collect. This portion of a CWS rule is called the Agent expression. Agent expressions use Datadog's Security Language (SECL). The standard format of a SECL expression is as follows:
```

Comment (sarina-dd) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md:13]: Combined both edits

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**